crates/aptos-ledger/src/lib.rs

```diff
@@ -0,0 +1,260 @@
+                Ok(hex_string)
+            } else {
+                let error_string = response
+                    .error_code()
+                    .map(|error_code| error_code.to_string())
+                    .unwrap_or_else(|retcode| format!("Error with retcode: {:x}", retcode));
+                Err(AptosLedgerError::UnexpectedError(error_string))
+            }
+        },
+        Err(err) => Err(AptosLedgerError::UnexpectedError(err.to_string())),
+    }
+}
+
+pub fn sign_txn(raw_txn: Vec<u8>) -> Result<Vec<u8>, AptosLedgerError> {
+    // open connection to ledger
+    let transport = match open_ledger_transport() {
+        Ok(transport) => transport,
+        Err(err) => return Err(err),
+    };
+
+    // serialize the derivative path
+    let derivative_path_bytes = serialize_bip32(DERIVATIVE_PATH);
+
+    // send the derivative path over as first message
+    let sign_start = transport.exchange(&APDUCommand {
+        cla: CLA_APTOS,
+        ins: INS_SIGN_TXN,
+        p1: P1_START,
+        p2: P2_MORE,
+        data: derivative_path_bytes,
+    });
+
+    if let Err(err) = sign_start {
+        return Err(AptosLedgerError::UnexpectedError(err.to_string()));
+    }
+
+    let chunks = raw_txn.chunks(MAX_APDU_LEN);
+    let chunks_count = chunks.len();
+
+    for (i, chunk) in chunks.enumerate() {
+        let is_last_chunk = chunks_count == i + 1;
+        match transport.exchange(&APDUCommand {
+            cla: CLA_APTOS,
+            ins: INS_SIGN_TXN,
+            p1: (i + 1) as u8,
+            p2: if is_last_chunk { P2_LAST } else { P2_MORE },
+            data: chunk.to_vec(),
+        }) {
+            Ok(response) => {
+                // success response
+                if response.retcode() == APDU_CODE_SUCCESS {
+                    if is_last_chunk {
+                        let response_buffer = response.data();
+
+                        let signature_len: usize = response_buffer[0] as usize;
+                        let signature_buffer = &response_buffer[1..1 + signature_len];
+                        return Ok(signature_buffer.to_vec());
+                    }
+                } else {
+                    let error_string = response
+                        .error_code()
+                        .map(|error_code| error_code.to_string())
+                        .unwrap_or_else(|retcode| {
+                            format!("Unknown Ledger APDU retcode: {:x}", retcode)
+                        });
+                    return Err(AptosLedgerError::UnexpectedError(error_string));
+                }
+            },
+            Err(err) => return Err(AptosLedgerError::UnexpectedError(err.to_string())),
+        };
+    }
+    Err(AptosLedgerError::UnexpectedError(
+        "Unable to process request".to_owned(),
+    ))
+}
+
+/// This is the Rust version of the serialization of BIP32 from Petra Wallet
+/// https://github.com/aptos-labs/wallet/blob/main/apps/extension/src/core/ledger/index.ts#L47
```

Comment:
If linking to code in another repo, don't point to main but to a specific rev, otherwise this link might break down the line.